packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -3051,10 +3051,10 @@ describe("Teal Opcodes", function () {
 			});
 
 			it("Tealv6: CallerApplicationID", function () {
-				// caller app id = 2
+				// caller app id = 1
 				const op = new Global(["CallerApplicationID"], 1, interpreter);
 				op.execute(stack);
-				assert.equal(2n, stack.pop());
+				assert.equal(1n, stack.pop());
```

Comment:
The problem is that the current application id is not pushed to the `innerTxAppIDCallStack`. Getting the second last element makes more sense when `currentAppId` was pushed to the call stack, or otherwise we should still use the last one and remove the action of `push`.
Here, the last element of `interpreter.runtime.ctx.innerTxAppIDCallStack = [1, 2];` was not coherent with `CurrentAppID`. Something is still wrong. (tho tests all pass)

Reply:
Actually this one is for unit test. Since you see that, we set up the inner transaction application call ID in [here](https://github.com/scale-it/algo-builder/pull/809/files#diff-0bd11169d40e3be81fef88fe8a8e4cf0795255b19d1f4d44d615ea872a1e0e90R3039). So the comment here https://github.com/scale-it/algo-builder/pull/798#issuecomment-1268772011 is where I misunderstand you

Reply:
Oh, ok. That's what I was thinking. Then your fix is what I was trying (but failed) to do. Congrats on finishing it 👍

Reply:
I'm not the creator not have write access in this repo. So you'd have to resolve these conversations by yourself @thdailong